x-pack/plugins/maps/public/classes/layers/layer.tsx

```diff
@@ -101,8 +102,10 @@ export interface ILayer {
   isIncludeInFitToBounds(): boolean;
   getLicensedFeatures(): Promise<LICENSED_FEATURES[]>;
   getCustomIconAndTooltipContent(): CustomIconAndTooltipContent;
+  queryForTileMeta(mbMap: MbMap): Feature[] | null;
```

Comment:
How removing "queryForTileMeta" from ILayer interface? Then just use casting in MbMap to only call for TiledVectorLayers? ILayer is already so cluttered. We need to strive to only add methods that are really used by multiple implementations.